proposals/0383-allow-let-property-wrapper.md

```diff
@@ -0,0 +1,179 @@
+}
+```
+
+A `let` wrapped property could be useful for reference types like a property wrapper class. Typically property wrappers are written for value types but occasionally a protocol like `NSObject` may require the use of a class. For example:
+```
+@propertyWrapper
+class WrapperClass<T> : NSObject {
+  var wrappedValue: T
+
+  init(wrappedValue: T) {
+    self.wrappedValue = wrappedValue
+  }
+}
+
+class C {
+  @WrapperClass let value: Int
+
+  init(v: Int) {
+    value = v
+  }
+}
+```
+Here, `WrapperClass` can be made an immutable property wrapper class instance, preventing any future unintentional changes to the property wrapper class type in this context.
+
+SwiftUI property wrappers may also benefit from a let declaration. For example, `@ScaledMetric` in its simplest usage can be written with a `let` instead:
+```
+struct ContentView: View {
+  @ScaledMetric let imageSize = 10
+
+  var body: some View {
+    Image(systemName: "heart.fill")
+    .resizable()
+    .frame(width: imageSize, height: imageSize)
+  }
+}
+```
+
+Similarly, other SwiftUI property wrappers could be `let` declared when they do not require more than a single initialization.
+
+### Property wrappers with `nonmutating set`
+
+Property wrappers that have a `wrappedValue` property with a `nonmutating set` (e.g., SwiftUI's [`@State`](https://developer.apple.com/documentation/swiftui/state/wrappedvalue) and [`@Binding`](https://developer.apple.com/documentation/swiftui/binding/wrappedvalue)) will preserve the reference semantics of the `wrappedValue` implementation even when marked as a `let` declaration. For example:  
+
+```
+@State let weekday: String = "Monday"
+```
+Here, `weekday` is an immutable instance of the `@State` property wrapper, but its `wrappedValue` storage will retain its mutability and reference type traits. This will translate to:
+```
+private let _weekday: State<String> = State<String>(wrappedValue: "Monday")
+var weekday : String {
+  get { 
+    return _weekday.wrappedValue 
+  }
+  nonmutating set {
+    self._weekday.wrappedValue = value
+  }
+  nonmutating _modify { 
+    yield () 
+  }
+}
+var $weekday: Binding<String> {
+  get { return _weekday.projectedValue }
+}
+```
+
+Marking `weekday` as a let declared property will not remove access to its `wrappedValue`'s nonmutating set and the `wrappedValue` can be assigned via the backing property, `_weekday`:
+
+```
+_weekday.wrappedValue = "Tuesday"
+```
+
+However, this does not affect the immutability of `weekday`, which can only be assigned to once like any ordinary let wrapped property. Any attempt to reassign `weekday` will result in an error:
+
+```
+weekday = "Wednesday" // Error: Cannot assign to value: 'weekday' is a 'let' constant
+```
+
+## Source compatibility
+
+This is an additive feature that does not impact source compatibility.
+
+## Effect on ABI stability/API resilience
+
+This is an additive change that has no direct impact that compromises ABI stability.
```

Comment:
In other reviews (maybe not this one), users have asked if new features can be backdeployed on older runtimes. It might be sufficiently obvious because property wrappers are a syntactic transformation, but for consistency it would be nice to comment here that no runtime support is needed for backdeployment.